tests/js/specs/row-actions.spec.js

```diff
@@ -34,36 +27,59 @@
         $('.letterhead-delete').click();
       });
 
-      it('displays a deleting letterhead message', () => {
-        expect(crmStatus).toHaveBeenCalledWith(
-          jasmine.objectContaining({ start: ts('Deleting...') }),
-          jasmine.any(Object)
-        );
+      it('displays a confirmation dialog', () => {
+        expect(crmConfirm).toHaveBeenCalledWith({
+          title: ts('Warning'),
+          message: ts('Are you sure you would like to delete this letterhead?'),
+          options: {
+            yes: ts('Yes'),
+            no: ts('Cancel')
+          }
+        });
       });
 
-      it('deletes the letterhead', () => {
-        expect(crmApi).toHaveBeenCalledWith('Letterhead', 'delete', {
-          id: letterheadId
+      describe('when the dialog is confirmed', () => {
+        beforeEach(() => {
+          crmConfirm.$dialog.trigger('crmConfirm:yes');
         });
-      });
 
-      describe('after deleting the letterhead', () => {
-        beforeEach((done) => {
-          crmStatus.resolve();
-          setTimeout(done);
+        /**
+         * @note The last parameter sent to *crmApi* is sent to the *crmStatus* service.
+         * If there is no `start` option for that parameter, a default "Saving" text will
+         * be used. This is the expected behaviour according to the specs.
+         */
+        it('displays a deleting letterhead message', () => {
+          expect(crmApi).toHaveBeenCalledWith(
```

Comment:
Not sure about this test? Why cant we check the parameters, and how this test is different from the one below?

Reply:
We can test that we are not sending the parameter is it's ok with you:

```js
expect(crmApi.calls.mostRecent().args[3].start).not.toBeDefined();
```

This is testing that we are not passing the `start` attribute of the last parameter so it displays "Saving".

The test below is checking that we are sending "Letterhead deleted" for the `success` attributes.